marimo/_sql/engines.py

```diff
@@ -76,19 +81,39 @@ def dialect(self) -> str:
     def execute(self, query: str) -> Any:
         # Can't use polars.imported() because this is the first time we
         # might come across polars.
+        if not (
+            DependencyManager.polars.has() or DependencyManager.pandas.has()
+        ):
+            raise_df_import_error("polars")
+
+        if DependencyManager.sqlalchemy.has():
+            from sqlalchemy import text
+
+            with self._engine.connect() as connection:
+                result = connection.execute(text(query))
+                connection.commit()
+
+            if not result.returns_rows:
+                return None
+
+            if DependencyManager.polars.has():
+                import polars as pl
+
+                return pl.DataFrame(result)  # type: ignore
+            else:
+                import pandas as pd
+
+                return pd.DataFrame(result)
+
+        # Fallback for non-SQLAlchemy execution, TODO: maybe suggest install sqlalchemy when fail
```

Comment:
maybe we make this into another "engine", based on what connection's `pl.read_database(` supports